lib/modules/manager/github-actions/schema.ts

```diff
@@ -8,7 +8,9 @@ import {
 
 const StepsSchema = z.object({
   uses: z.string(),
-  with: LooseRecord(z.string()),
+  with: LooseRecord(
+    z.union([z.string(), z.number().transform((s) => s.toString())]),
+  ),
```

Comment:
```suggestion
  ).optional(),
```
🤔  should `with` optional here?

Reply:
not really, we do not use it yet

Reply:
Thanks -- as @viceice mentioned, seems like this PR will also solve the issue described in #36129 when the version is a float or int, not a string.

I scanned this quickly and didn't see a test for a case like that - would it make sense to add a test here (see diff in https://github.com/renovatebot/renovate/discussions/36129#discussioncomment-13265736) to cover the case where the input version is a number? And might be good to call out this fix in the description and / or break out this update first separately)?